### PR TITLE
Feature/526 update background submissions migration

### PR DESCRIPTION
**Pull request by kreek**
## Background
526 form submissions are moving from synchronous to asynchronous requests within a background job. As the background work is out of the request response cycle we need store the response somewhere so that the FE can poll for it later.

## Definition of Done
Update the `disability_compensation_submissions` table with columns for the response body (that stores the entire response rather than just claim_id), status of the job, and job id.

#### Unique to this PR

- [x] Stores the response on success and failure
- [x] Stores the job id
- [x] Stores the status of the job

#### Applies to all PRs

- [x] Appropriate test coverage & logging
- [x] Swagger docs have been updated, if applicable
- [x] Provide link to originating GitHub issue, or connected to it via ZenHub
- [x] Does not contain any sensitive information (i.e. PII/credentials/internal URLs/etc., in logging, hardcoded, or in specs)

### CODE REVIEW

File: app/models/disability_compensation_submission.rb [hunk 1]
```diff
@@ -1,7 +1,15 @@
...
+    received: 'received',
+    retrying: 'retrying',
+    non_retryable_error: 'non_retryable_error',
+    exhausted: 'exhausted'
+  }
+
   validates :user_uuid, presence: true, uniqueness: { scope: :form_type }
   validates :form_type, presence: true, uniqueness: { scope: :user_uuid }
-  validates :claim_id, presence: true, uniqueness: true
+  validates :status, presence: true
```
Comment (egarreau) [inline app/models/disability_compensation_submission.rb]: do we need to validate the presence of `job_id` and `response` as well?

Comment (kreek) [reply app/models/disability_compensation_submission.rb]: Good catch, yes for `job_id`, response can be null when the status is `submitted`

**Comment by kreek** [comment]
@SMLuthi 🤔seems like we could, do you know if transaction id maps to job id, is the metadata for the response?

**Comment by kreek** [comment]
Closing this, I like @SMLuthi 's idea of piggy-backing off the async transactions better.

**Pull request closed without merging**